docs/cupynumeric/source/user/Legate Task tutorial.rst

```diff
@@ -0,0 +1,981 @@
+    end = time()
+    
+    print(f"\nTime elapsed for saxpy: {(end - start)/1000:.6f} milliseconds")
+
+
+The Legate runtime is used in the main function to control and
+synchronize task execution. The ``get_legate_runtime()`` function returns
+this runtime, which is used to issue commands like execution fences. In
+this example, ``issue_execution_fence()`` is called before and after the
+``saxpy_task`` to ensure accurate time measurement. Since Legate tasks run
+asynchronously by default, these fences make the program wait until all
+previous tasks have finished, so the measured time reflects only the
+actual task execution. This is a common pattern when precise timing,
+synchronization, or ordered execution of asynchronous tasks is needed.
+
+GPU initialization has a fixed setup time that can significantly affect
+the runtime when processing small arrays. Using a large input (e.g., 100
+million elements) ensures that the computation time outweighs the
+startup overhead, giving more realistic timing results. Since the first
+GPU run may include the setup overhead like compilation or memory
+allocation, a warm-up pass helps eliminate these one-time costs from
+performance measurements, ensuring more reliable results.
+
+
+Running on CPU and GPU 
+----------------------
+
+In order to run the program, use the legate launcher, and include any
+flags necessary like ``--cpus``, ``--gpus``, and more. If you want to run
+specifically only on CPU, you must include the flag ``--gpus 0``.
+For a complete guide and additional options, see the `Legate documentation`_.
+
+.. _Legate documentation: https://docs.nvidia.com/legate/latest/usage.html
+
+Let’s set the input array size to 100 million elements to better
+evaluate the speedup from distributed computing with GPUs.
+
+
+CPU execution 
+~~~~~~~~~~~~~
+
+To run with CPU, use the following command.
+
+.. code-block:: sh
+
+    legate --cpus 1 --gpus 0 ./saxpy.py --size 100000000
+
+This produces the following output:
+
+.. code-block:: text
+
+    Time elapsed for saxpy: 146.303000 milliseconds
+
+GPU execution 
+~~~~~~~~~~~~~
+
+To run with GPU, use the following command.
+
+.. code-block:: sh
+
+    legate --gpus 2 ./saxpy.py --size 100000000
+
+This produces the following output:
+
+.. code-block:: text
+
+    Time elapsed for saxpy : 1.949000 milliseconds
+
+Multi-Node execution 
+~~~~~~~~~~~~~~~~~~~~
+Refer to the Legate documentation on how to run on `multi-node`_. 
+Here is an example performed on the `Perlmutter`_ supercomputer.
+
+.. _multi-node: https://docs.nvidia.com/legate/latest/usage.html
+.. _Perlmutter: https://docs.nersc.gov/systems/perlmutter/architecture/
+
+To run on multi-node, use the following command.
+
+.. code-block:: sh
+
+    legate --nodes 2 --launcher srun --gpus 4 --ranks-per-node 1 ./saxpy.py --size 100000000
+
+This produces the following output:
+
+.. code-block:: text
+
+    Time elapsed for saxpy : 2.052000 milliseconds
+
+Histogram problem
+=================
```

Comment:
This should be a separate rst file. In general, all of the separate problems described here should be a separate rst.

Reply:
These are meant to be examples of using Legate tasks, building up in complexity. I think it's better to keep them all in the same page as separate sections, rather than having them as separate pages. This is how the other tutorial is set up for example.

That's just my take though, would ask for @dongb and @bryevdv to weigh in too.

Reply:
I'd keep things in one file for now, but I'd reiterate the suggestion to re-organize along these lines:
```
tutorial\
    basic.rst  # this is the existing tutorial.rst
    task.rst.  # this is the task tutorial added in this PR
    index.rst  # landing page for all tutorials that links to basic and task, etc. 
```